lib/node.js

```diff
@@ -1,178 +1,171 @@
 "use strict";
 
-var P = require('bluebird');
+const P = require('bluebird');
+
+// Work around recursive structure in ** terminal nodes
+function printableValue(value) {
+    const res = {};
+    if (!value || !(value instanceof Object)) {
+        return value;
+    }
+    Object.keys(value).forEach((key) => {
+        const val = value[key];
+        if (key === 'methods') {
+            const newMethods = {};
+            Object.keys(val).forEach((method) => {
+                newMethods[method] = `<${val[method].name}>`;
+            });
+            res.methods = newMethods;
+        } else if (key !== 'specRoot') {
+            // Omit the specRoot, as it tends to be huge & contains reference
+            // circles.
+            res[key] = val;
+        }
+    });
+    return res;
+}
 
 /*
  * A node in the lookup graph.
  *
  * We use a single monomorphic type for the JIT's benefit.
  */
-function Node(value) {
-    // The value for a path ending on this node. Public property.
-    this.value = value || null;
-
-    // Internal properties.
-    this._children = {};
-    this._paramName = null;
-    this._parent = null;
-}
-
-Node.prototype._keyPrefix = '/';
-Node.prototype._keyPrefixRegExp = /^\//;
+class Node {
+    constructor(value) {
+        // The value for a path ending on this node. Public property.
+        this.value = value || null;
+
+        // Internal properties.
+        this._children = {};
+        this._paramName = null;
+        this._parent = null;
+
+        this._keyPrefix = '/';
```

Comment:
This makes the keyprefix & -regexp a property of the instance, rather than the prototype. Those prefixes should be const, so we could perhaps juts define them outside as a module-global const. Alternatively, we could say `Node._keyPrefix = '/';` after the class definition, which defines it as a prototype method.